server/routes/createGoal/index.ts

```diff
@@ -52,18 +52,15 @@ export default (router: Router, services: Services) => {
     createGoalController.submitAddNoteForm,
   ])
 
+  router.use('/plan/:prisonNumber/goals/review', checkUserHasEditAuthority())
```

Comment:
Not really related to this PR, but the security check was missing for this route, so I added it